src/js/actions/export.js

```diff
@@ -358,23 +359,41 @@ define(function (require, exports) {
     exportAllAssets.transfers = [updateLayerExportAsset];
 
     /**
-     * before start up, initialize the export service
+     * Before start up, Ensure that generator is enabled, and then initialize the export service
      * @return {Promise}
      */
     var beforeStartup = function () {
-        _exportService = new ExportService();
-        return _exportService.init()
+        return descriptor.playObject(generatorLib.getGeneratorStatus())
             .bind(this)
-            .then(function () {
-                return this.dispatchAsync(events.export.SERVICE_STATUS_CHANGED, { serviceAvailable: true });
+            .then(function (status) {
+                var enabled = status.generatorStatus.generatorStatus === 1;
+                if (!enabled) {
+                    log.info("Enabling Generator...");
+                    return descriptor.playObject(generatorLib.setGeneratorStatus(true))
+                        .catch(function (e) {
+                            throw new Error("Could not enable generator", e);
+                        });
+                }
             })
-            .catch(Promise.TimeoutError, function (e) {
-                log.error("Could not connect to generator plugin!", e);
-                return Promise.resolve("Export Service not enabled, but giving up because of timeout");
+            .delay(500)
```

Comment:
After generator startup (well, after the response to the photoshop command), it takes some time before the plugin is initialized and the websocket will accept requests.  Even after this 500ms delay, you'll notice that it takes another few seconds before the websocket connections don't fail.  Maybe there is a more positive way to do this (maybe get an event from Ps when generator has started?)